components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/GatewayArtifactsMgtDBUtil.java

```diff
@@ -0,0 +1,167 @@
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class GatewayArtifactsMgtDBUtil {
+
+    private static final Log log = LogFactory.getLog(GatewayArtifactsMgtDBUtil.class);
+    private static volatile DataSource artifactSynchronizerDataSource = null;
+    private static final String DATA_SOURCE_NAME = "DataSourceName";
+
+    /**
+     * Initializes the data source
+     *
+     * @throws APIManagementException if an error occurs while loading DB configuration
+     */
+    public static void initialize() throws APIManagerDatabaseException {
+        if (artifactSynchronizerDataSource != null) {
+            return;
+        }
+
+        synchronized (APIMgtDBUtil.class) {
+            if (artifactSynchronizerDataSource == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Initializing data source");
+                }
+                GatewayArtifactSynchronizerProperties gatewayArtifactSynchronizerProperties =
+                        ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                                .getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+                String artifactSynchronizerDataSourceName =
+                        gatewayArtifactSynchronizerProperties.getArtifactSynchronizerDataSource();
+
+                if (artifactSynchronizerDataSourceName != null) {
+                    try {
+                        Context ctx = new InitialContext();
+                        artifactSynchronizerDataSource = (DataSource) ctx.lookup(artifactSynchronizerDataSourceName);
+                    } catch (NamingException e) {
+                        throw new APIManagerDatabaseException("Error while looking up the data " +
+                                "source: " + artifactSynchronizerDataSourceName, e);
+                    }
+                } else {
+                    log.error("AS" + artifactSynchronizerDataSourceName + " not defined in api-manager.xml.");
+                }
+            }
+        }
+    }
+
+    /**
+     * Utility method to get a new database connection for gatewayRuntime artifacts
+     *
+     * @return Connection
+     * @throws java.sql.SQLException if failed to get Connection
+     */
+    public static Connection getArtifactSynchronizerConnection() throws SQLException {
+        if (artifactSynchronizerDataSource != null) {
+            return artifactSynchronizerDataSource.getConnection();
+        }
+        throw new SQLException("Data source is not configured properly.");
+    }
+
+    /**
+     * Utility method to close the connection streams.
+     * @param preparedStatement PreparedStatement
+     * @param connection Connection
+     * @param resultSet ResultSet
+     */
+    public static void closeAllConnections(PreparedStatement preparedStatement, Connection connection,
```

Comment:
do we need this

Reply:
Removed it . https://github.com/wso2/carbon-apimgt/pull/8828/commits/d501401001c713712bb50c42b40aa2afa9326545